src/GuiBase/Viewer/TrackballCamera.cpp

```diff
@@ -161,6 +154,79 @@ bool Gui::TrackballCamera::handleKeyReleaseEvent( QKeyEvent* e ) {
     return false;
 }
 
+void Gui::TrackballCamera::setCamera( Engine::Camera* camera ) {
+    if ( !camera )
+        return;
+    m_camera = camera;
+    m_trackballCenter = m_camera->getPosition() + 2 * m_camera->getDirection().normalized();
+    m_distFromCenter = 2.0;
+    updatePhiTheta();
+    m_camera->show( false );
+}
+
+void Gui::TrackballCamera::save( std::ostream& out ) const {
+    out << "#Radium_camera_state" << std::endl;
+    out << (int)m_camera->getType() << std::endl;
+    out << m_camera->getFrame().matrix() << std::endl;
+    out << std::endl;
+    out << m_camera->getFOV() << " " << m_camera->getZNear() << " " << m_camera->getZFar() << " "
+        << m_camera->getZoomFactor() << " " << m_camera->getAspect() << std::endl;
+    out << std::endl;
+    out << m_trackballCenter.transpose() << " " << m_distFromCenter << std::endl;
+}
+
+void Gui::TrackballCamera::load( std::istream& in ) {
+    std::string str;
```

Comment:
Why do we still have load/save cameras out of the loader ?

Reply:
Because it is for the UI.
The Engine::Camera is a Component, which can be loaded from assets.
The Viewer::CameraInterface is a Camera manipulator, with its own additional parameters depending on the manipulator type (e.g. trackball has a rotation center).

Reply:
I see.

Reply:
I agree. See my comment above on CameraLoader. If this is a functionnality needed by a single plugin, let the plugin manage direct loading of camera, after adding a widget in the main-app ... If this functionality is needed by several plugins, then, make a didicated camera plugin that extends the main app only with the .cam file loader and the widget that allow to control the loaded camera.

Reply:
I do believe loading and saving cameras could be a great addition to RadiumIO. As add a camera loader is not very intrusive, but the gain very high, I suggest we skip the plugin step and add the loader in IO. 
The custom load and save functions however have nothing to do in Radium code. The application can ask the loader to save the camera. And the camera manipulator must hold a camera, isn’t ?